packages/grafana-data/src/transformations/transformers/filter.ts

```diff
@@ -22,9 +22,25 @@ export const filterFieldsTransformer: DataTransformerInfo<FilterOptions> = {
    * Return a modified copy of the series.  If the transform is not or should not
    * be applied, just return the input series
    */
-  operator: (options: FilterOptions) => (source) => {
+  operator: (options: FilterOptions, replace) => (source) => {
     if (!options.include && !options.exclude) {
-      return source.pipe(noopTransformer.operator({}));
+      return source.pipe(noopTransformer.operator({}, replace));
+    }
+
+    if (typeof options.include?.options === 'string') {
+      options.include.options = replace ? replace(options.include?.options) : options.include?.options;
+    } else if (typeof options.include?.options?.pattern === 'string') {
+      options.include.options.pattern = replace
+        ? replace(options.include?.options.pattern)
+        : options.include?.options.pattern;
+    }
+
+    if (typeof options.exclude?.options === 'string') {
+      options.exclude.options = replace ? replace(options.exclude?.options) : options.exclude?.options;
+    } else if (typeof options.exclude?.options?.pattern === 'string') {
+      options.exclude.options.pattern = replace
+        ? replace(options.exclude?.options.pattern)
+        : options.exclude?.options.pattern;
```

Comment:
This is slightly hard to read, could we change it to have the `replace` as a condition at the top?
```
    if (replace) {
      if (typeof options.include?.options === 'string') {
        options.include.options = replace(options.include?.options);
      } else if (typeof options.include?.options?.pattern === 'string') {
        options.include.options.pattern = replace(options.include?.options.pattern);
      }

      if (typeof options.exclude?.options === 'string') {
        options.exclude.options = replace(options.exclude?.options);
      } else if (typeof options.exclude?.options?.pattern === 'string') {
        options.exclude.options.pattern = replace(options.exclude?.options.pattern);
      }
    }
```